src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,79 @@
+package parser;
+
+import java.time.format.DateTimeParseException;
+
+import command.AddCommand;
+import command.Command;
+import command.DeleteCommand;
+import command.ExitCommand;
+import command.InvalidCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.UnmarkCommand;
+import task.Deadline;
+import task.Event;
+import task.ToDo;
+
+/**
+ * Represents a parser to parse user input.
+ */
+public class Parser {
+    /**
+     * Parses the given input and returns the corresponding command.
+     *
+     * @param input The input to parse.
+     * @return The corresponding command.
+     */
+    public static Command parse(String input) {
+        String[] parts = input.split(" ", 2);
+        String command = parts[0];
+        String arguments = parts.length > 1 ? parts[1] : "";
+
+        switch (command) {
+        case "list":
+            return new ListCommand();
+        case "todo":
+            return new AddCommand(new ToDo(arguments));
+        case "deadline":
```

Comment:
Again a matter of opinion. Maybe the exception handling could be done in the individual command classes?  `Parser` could still handle the splitting and arguments though.